src/admin/components/extensive/Modal/ModalApprove.tsx

```diff
@@ -28,9 +32,27 @@ export interface ModalApproveProps extends ModalProps {
 interface DisplayedPolygonType {
   id: string | undefined;
   name: string | undefined;
+  checked: boolean | undefined;
   canBeApproved: boolean | undefined;
   failingCriterias: string[] | undefined;
 }
+type ValidationCriteria = GetV2TerrafundValidationSiteResponse[number];
+
+const checkCriteriaCanBeApproved = (criteria: ValidationCriteria) => {
+  if (!criteria.checked) {
+    return false;
+  }
+  if (criteria?.nonValidCriteria?.length === 0) {
+    return true;
+  }
+  const excludedFromValidationCriterias = [COMPLETED_DATA_CRITERIA_ID, ESTIMATED_AREA_CRITERIA_ID];
+  const nonValidCriteriasIds = criteria?.nonValidCriteria?.map(r => r.criteria_id);
+  const failingCriterias = nonValidCriteriasIds?.filter(r => !excludedFromValidationCriterias.includes(r));
+  if (failingCriterias?.length === 0) {
+    return true;
+  }
+  return false;
```

Comment:
These lines can be simplified to `return failingCriterias?.length === 0`

Reply:
changed